src/main/app/src/pages/toteutus/ToteutusForm/TuvaTelmaTiedotSection.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect } from 'react';
+
+import { useTranslation } from 'react-i18next';
+import { Field } from 'redux-form';
+
+import { FormFieldInput, FormFieldSwitch } from '#/src/components/formFields';
+import { Box } from '#/src/components/virkailija';
+import { KOULUTUSTYYPPI } from '#/src/constants';
+import { useBoundFormActions } from '#/src/hooks/form';
+import useKoodi from '#/src/hooks/useKoodi';
+import { getTestIdProps } from '#/src/utils';
+import { getOpintojenLaajuusTranslations } from '#/src/utils/getOpintojenLaajuusTranslations';
+
+export const TuvaTelmaTiedotSection = ({ language, name, koulutus }) => {
+  const { t } = useTranslation();
+
+  const { koodi: laajuusKoodi } = useKoodi(
+    koulutus.metadata.opintojenLaajuusKoodiUri
+  );
+  const { koodi: laajuusyksikko } = useKoodi('opintojenlaajuusyksikko_6');
+  const laajuusKoodiNimet = laajuusKoodi?.metadata;
+  const laajuusyksikkoNimet = laajuusyksikko?.metadata;
+
+  const koulutustyyppi = koulutus.koulutustyyppi;
+  const { change } = useBoundFormActions();
+  useEffect(() => {
+    change(`${name}.nimi`, {
+      fi: t(`koulutustyypit.${koulutustyyppi}`, { lng: 'fi' }),
+      sv: t(`koulutustyypit.${koulutustyyppi}`, { lng: 'sv' }),
+      en: t(`koulutustyypit.${koulutustyyppi}`, { lng: 'en' }),
+    });
+    return () => change(`${name}.nimi`, {});
+  }, [change, koulutustyyppi, name, t]);
+
+  useEffect(() => {
+    change(
+      `${name}.laajuus`,
+      getOpintojenLaajuusTranslations(laajuusKoodiNimet, laajuusyksikkoNimet)
+    );
+    return () => change(`${name}.laajuus`, {});
+  }, [change, laajuusKoodiNimet, laajuusyksikkoNimet, name, t]);
```

Comment:
Laajuus ei taida olla arvo, jota tarvisi tallentaa toteutukselle, joten sinänsä varmaan turhaa kierrättää sitä reduxin kautta näin.

Reply:
Korjasin tämän nyt ja testasin QA:lla.